ferrocene/doc/requirements/src/self-test.rst

```diff
@@ -0,0 +1,209 @@
+.. SPDX-License-Identifier: MIT OR Apache-2.0
+   SPDX-FileCopyrightText: The Ferrocene Developers
+
+`ferrocene-self-test` tool
+==========================
+
+General checks
+--------------
+
+Checks for executables
+^^^^^^^^^^^^^^^^^^^^^^
+
+.. note::
+
+   **executable** refers to the following:
+
+   - ``rustc``
+   - ``rustdoc``
+
+The executable should exist in the ``bin`` directory, relative to the root of the installation.
+
+The metadata of the executable should be fetchable.
+
+The executable should be a regular file.
+
+The executable should have read and execute permissions set for all users.
+
+When executed with ``-vV`` command-line options, the following apply:
+
+- The executable should exit successfully.
+
+- The output of the executable should be UTF-8 text.
+
+- The output of the executable should have at least 3 values: host, commit-hash, and release.
```

Comment:
What is the syntax of the output? What is "values" in this context?

Reply:
b956c2915a1